src/url.rs

```diff
@@ -150,6 +154,46 @@ impl PyUrl {
     fn __getnewargs__(&self) -> (&str,) {
         (self.__str__(),)
     }
+
+    #[classmethod]
+    #[pyo3(signature=(*, scheme, host, user=None, password=None, port=None, path=None, query=None, fragment=None))]
+    #[allow(clippy::too_many_arguments)]
+    pub fn build<'a>(
+        cls: &'a PyType,
+        scheme: &str,
+        host: &str,
+        user: Option<&str>,
+        password: Option<&str>,
+        port: Option<&str>,
```

Comment:
Similarly we may want to correct port to be an `int` both here and below for `MultiHostUrl.build`?